docs/tutorial/creating-new-tools.md

```diff
@@ -0,0 +1,154 @@
+
+The `AbstractTool` class requires you to implement a `run` method. This method is
+where you will implement the core functionality of your tool. The `run` method will
+be called when your tool is run. 
+
+The `run` method takes a single argument `tool_input`, which is an instance of the 
+`ToolInput` class. The `ToolInput` class is defined in the `geneweaver.tools.framework`
+module, and is a subclass of the `Pydantic` `BaseModel` class. The `ToolInput` class
+is used to define the input data model for your tool.
+
+The `run` method must return an instance of the `ToolOutput` class. The `ToolOutput`
+class is defined in the `geneweaver.tools.framework` module, and is a subclass of the
+`Pydantic` `BaseModel` class. The `ToolOutput` class is used to define the output data
+model for your tool.
+
+This means that the abstract method signature for the `run` method is as follows:
+
+```python
+@abstractmethod
+def run(self: AbstractTool, tool_input: ToolInput) -> ToolOutput:
+    """Run the tool."""
+```
+
+#### Input and Output Abstract Properties
+Your tool class must also implement two abstract properties: `input` and`output`. These 
+properties are used to define the input and output data models, and must return the
+uninstantiated `ToolInput` and `ToolOutput` classes, respectively.
+
+The abstract method signatures for the `input` and `output` properties are as follows:
+
+```python
+@property
+@abstractmethod
+def input(self: AbstractTool) -> Type[ToolInput]:
+    """Input schema for the tool."""
+
+@property
+@abstractmethod
+def output(self: AbstractTool) -> Type[ToolOutput]:
+    """Output schema for the tool.""" 
+```
+
+#### Overrideable Properties
+Your tool class can also implement a few overrideable properties. These properties
+are used to define the metadata for your tool, and are used to display information
+about your tool in the Geneweaver web application.
+
+- `tool_name`: The name of your tool. This is used to display the name of your tool
+  in the Geneweaver web application. By default, this is set to the name of your
+  tool class.
+- `static_files_location`: The location of the static files for your tool. This is
+  used to load static data like images and CSS files for your tool in the Geneweaver
+  web application. By default, this is set to the `static` directory in the same
+  directory as your tool class.
+
+##### Tool Workflow Definition
+
+!!! tip "Scientific Workflows"
+    A scientific workflow is a series of interconnected tasks or computational steps 
+    that are designed to solve a specific problem or address a scientific question. Not
+    all analysis tools will need to define a scientific workflow, and for single-step
+    tools, this property can be left undefined as it can be inferred by the Geneweaver
+    web application. 
+
+    However, complex analysis tools are encouraged to define a scientific workflow.
+
+    Read more about how Geneweaver uses scientific workflows to define analysis tools
```

Comment:
about "how" you define? I didn't see the "how" above. Is that the page where you want to dive into the weeds?